cancel_token/token.py

```diff
@@ -132,21 +143,35 @@ def cancel_not_done(fut: 'asyncio.Future[None]') -> None:
                 return_when=asyncio.FIRST_COMPLETED,
                 loop=self.loop,
             )
-        except asyncio.futures.CancelledError:
-            # Since we use return_when=asyncio.FIRST_COMPLETED above, we can be sure none of our
-            # futures will be done here, so we don't need to check if any is done before cancelling.
+        except asyncio.CancelledError as err:
+            # Since we use return_when=asyncio.FIRST_COMPLETED above, we can
+            # be sure none of our futures will be done here, so we don't need
+            # to check if any is done before cancelling.
             for future in futures:
                 future.cancel()
-            raise
-        for task in pending:
-            task.cancel()
+            for future in futures:
+                try:
+                    await future
+                except asyncio.CancelledError:
+                    pass
+            raise err
+        else:
+            for future in pending:
+                future.cancel()
+            for future in pending:
+                try:
+                    await future
+                except asyncio.CancelledError:
```

Comment:
4. This one fixes futures that aren't awaited in the most normal case: one of the awaitables completes (there will always be another one pending, because of the cancel token)

Reply:
this is now covered in `test_other_awaitables_are_cancelled_if_one_finishes`